packages/airnode-node/src/api/processing.ts

```diff
@@ -0,0 +1,84 @@
+import { Endpoint, ProcessingSpecification } from '@api3/airnode-ois';
+import { go } from '@api3/airnode-utilities';
+import { CallApiPayload } from './index';
+
+export const preProcessApiSpecifications = async (payload: CallApiPayload): Promise<CallApiPayload> => {
+  const { config, aggregatedApiCall } = payload;
+  const { endpointName, oisTitle } = aggregatedApiCall;
+  const ois = config.ois.find((o) => o.title === oisTitle)!;
+  const { preProcessingSpecifications } = ois.endpoints.find((e) => e.name === endpointName)!;
+
+  if (!preProcessingSpecifications || preProcessingSpecifications?.length === 0) {
+    return payload;
+  }
+
+  const [err, parameters] = await go(
+    async () =>
+      preProcessingSpecifications.reduce((input: any, currentValue: ProcessingSpecification) => {
+        switch (currentValue.environment) {
+          case 'Node 14':
+            return eval(`
+                const ethers = require('ethers');
```

Comment:
We are supplying `ethers` for the API providers to use, but this versions will keep changing (and the API provider will not know what version they are using unless we make it clear somehow). 

I don't think we need to resolve this now, but I'd add a note here that we need to think about and solve this somehow.

Reply:
You are right. Ideally we'd want a separate bespoke execution environment. This is just a stop-gap to iterate on.